tasks/upload.py

```diff
@@ -497,18 +504,116 @@ def run_impl_within_lock(
             )
             self.retry(countdown=60, kwargs=upload_context.kwargs_for_retry(kwargs))
 
+        upload_argument_list = self.possibly_insert_uploads_and_side_effects(
+            db_session=db_session,
+            upload_context=upload_context,
+            commit=commit,
+            commit_report=commit_report,
+            report_service=report_service,
+        )
+
+        if upload_argument_list:
+            db_session.commit()
+
+            UPLOADS_PER_TASK_SCHEDULE.labels(report_type=report_type.value).observe(
+                len(upload_argument_list)
+            )
+            scheduled_tasks = self.schedule_task(
+                commit,
+                commit_yaml.to_dict(),
+                upload_argument_list,
+                commit_report,
+                upload_context,
+            )
+
+            log.info(
+                f"Scheduling {upload_context.report_type.value} processing tasks",
+                extra=upload_context.log_extra(
+                    argument_list=upload_argument_list,
+                    number_arguments=len(upload_argument_list),
+                    scheduled_task_ids=scheduled_tasks.as_tuple(),
+                ),
+            )
+
+        else:
+            self.maybe_log_upload_checkpoint(UploadFlow.INITIAL_PROCESSING_COMPLETE)
+            self.maybe_log_upload_checkpoint(UploadFlow.NO_REPORTS_FOUND)
+            log.info(
+                "Not scheduling task because there were no arguments found on redis",
+                extra=upload_context.log_extra(),
+            )
+        return {"was_setup": was_setup, "was_updated": was_updated}
+
+    def possibly_insert_uploads_and_side_effects(
+        self,
+        db_session: Session,
+        upload_context: UploadContext,
+        commit: Commit,
+        commit_report: CommitReport,
+        report_service: ReportService,
+    ) -> list[UploadArguments]:
+        """
+        This method possibly batch inserts uploads, flags and user measurements.
```

Comment:
we know it "possibly" does something from the name. What are the conditions that the upload will or will not be inserted?

Reply:
It's described in the sentence after that, https://github.com/codecov/worker/pull/1006/files#diff-7ea195247ce946074a2441630db2be4bda78feed427fc0a1f1bd170cfc84a7a7R557 😅, only happens for v4 uploads since cli uploads are created in api. Should I rephrase it a bit so it's clearer?